Support ARM SVE via vector_size API

crtrott opened the pull request.

crtrott committed:
Support ARM SVE via vector_size API

crtrott commented on simd.hpp:
```diff
@@ -49,9 +49,7 @@
 
 #include "pack.hpp"
 
-#if defined(__clang_major__) && (__clang_major__ >= 11)
```
Moved the correct guard into the vector_size.hpp file

crtrott commented on simd.hpp:
```diff
@@ -97,12 +95,16 @@ using native = avx512;
 #elif defined(__SSE2__)
 using native = sse;
-#elif defined(__ARM_NEON)
+#elif defined(__ARM_NEON) && !defined(__ARM_FEATURE_SVE_BITS)
```
NEON is defined for SVE supported archs too, but we don't want to use it as the native type.

crtrott commented on simd.hpp:
```diff
@@ -97,12 +95,16 @@ using native = avx512;
 #else
-using native = pack<8>;
+#if defined(__ARM_FEATURE_SVE_BITS)
+using native = vector_size<__ARM_FEATURE_SVE_BITS/8>;
```
vector size takes the size in bytes.

crtrott commented on vector_size.hpp:
```diff
@@ -45,7 +45,8 @@
 
-#if defined(__clang__)
+#if (defined(__clang__) && (__clang_major__ >= 11)) || \
+    (defined(__GNUC__) && (__GNUC__ >= 10) && (__GNUC_MINOR__ >= 2))
```
GNU 10.2 supports this

DavidPoliakoff reviewed (approved):
Looks good to me, certainly supports the SVE that we know about

crtrott committed:
Fix build for Fujitsu and ArmClang

since armclang doesn't identify itself in code you can activate the
vector_size ABI by defining SIMD_ENABLE_VECTOR_SIZE explicitly
